h2/src/main/org/h2/value/Value.java

```diff
@@ -716,585 +716,588 @@ public Value convertTo(int targetType, int precision, Mode mode, Object column,
+        switch (getType()) {
+        case BYTES:
+        case BLOB:
+            return ValueJavaObject.getNoCopy(null, getBytesNoCopy(), getDataHandler());
+        case ENUM:
+        case TIMESTAMP_TZ:
+            throw getDataConversionError(JAVA_OBJECT);
+        }
+        return ValueJavaObject.getNoCopy(null, StringUtils.convertHexToBytes(getString().trim()), getDataHandler());
+    }
+
+    private ValueEnum convertToEnumInternal(String[] enumerators) {
+        switch (getType()) {
+        case BYTE:
+        case SHORT:
+        case INT:
+        case LONG:
+        case DECIMAL:
+            return ValueEnum.get(enumerators, getInt());
+        case STRING:
+        case STRING_IGNORECASE:
+        case STRING_FIXED:
+            return ValueEnum.get(enumerators, getString());
+        case JAVA_OBJECT:
+            Object object = JdbcUtils.deserialize(getBytesNoCopy(), getDataHandler());
+            if (object instanceof String) {
+                return ValueEnum.get(enumerators, (String) object);
+            } else if (object instanceof Integer) {
+                return ValueEnum.get(enumerators, (int) object);
+            }
+            //$FALL-THROUGH$
+        }
+        throw getDataConversionError(ENUM);
+    }
+
+    private ValueLobDb convertToBlob() {
+        switch (getType()) {
+        case BYTES:
+            return ValueLobDb.createSmallLob(Value.BLOB, getBytesNoCopy());
+        case TIMESTAMP_TZ:
+            throw getDataConversionError(BLOB);
+        }
+        return ValueLobDb.createSmallLob(BLOB, StringUtils.convertHexToBytes(getString().trim()));
+    }
+
+    private ValueLobDb convertToClob() {
+        return ValueLobDb.createSmallLob(CLOB, getString().getBytes(StandardCharsets.UTF_8));
+    }
+
+    private ValueUuid convertToUuid() {
+        switch (getType()) {
+        case BYTES:
+            return ValueUuid.get(getBytesNoCopy());
+        case JAVA_OBJECT:
+            Object object = JdbcUtils.deserialize(getBytesNoCopy(), getDataHandler());
+            if (object instanceof java.util.UUID) {
+                return ValueUuid.get((java.util.UUID) object);
+            }
+            //$FALL-THROUGH$
+        case TIMESTAMP_TZ:
+            throw getDataConversionError(UUID);
+        }
+        return ValueUuid.get(getString());
+    }
+
+    private ValueGeometry convertToGeometry() {
+        switch (getType()) {
+        case BYTES:
+            return ValueGeometry.get(getBytesNoCopy());
+        case JAVA_OBJECT:
+            Object object = JdbcUtils.deserialize(getBytesNoCopy(), getDataHandler());
+            if (DataType.isGeometry(object)) {
+                return ValueGeometry.getFromGeometry(object);
+            }
+            //$FALL-THROUGH$
+        case TIMESTAMP_TZ:
+            throw getDataConversionError(GEOMETRY);
+        }
+        return ValueGeometry.get(getString());
+    }
+
+    private ValueInterval convertToInterval1(int targetType) {
```

Comment:
May be `convertToIntervalYearMonth()` or `convertToYearMonthInterval()`?

“Year-month intervals” is a name of this class of intervals from the standard.